assignments/this.js

```diff
@@ -13,14 +13,49 @@
 
 // code example for Window Binding
 
+function hello(greeting){
+    console.log(this);
+    return greeting;
+}
+console.log(hello("hi"));
+
 // Principle 2
 
 // code example for Implicit Binding
 
+const hero = {
+    name: "spiderman",
+    nickname: "spidy",
+    alertDanger: function() { console.log(`My ${this.nickname} senses are going off!`)},
+};
+
+hero.alertDanger();
+
+
+
 // Principle 3
 
 // code example for New Binding
 
+function AveragePerson(name) {
+    this.name = name;
+    this.ask = function() {
+        console.log(`I'm ${this.name}. Have you seen my killer?`);
+    };
+}
+
+const johnDoe = new AveragePerson("John Doe");
+
+johnDoe.ask();
+
+
 // Principle 4
 
-// code example for Explicit Binding
\ No newline at end of file
+// code example for Explicit Binding
+
+const villain = {
```

Comment:
```suggestion
const villain = {
    name: "Otto",
    nickname: "Dr. Octopus",
}

function sayName() { return `Hello my name is ${this.name}` }

console.log(sayName.call(villain));
```
This is a better example for explicit binding the function will return `Hello my name is ${this.name}` but the function doesn't know know where this.name is until you bind it with .call(villain) which will tell the function that this will point towards the villain object